src/Compilers/CSharp/Test/Symbol/Symbols/Metadata/PE/LoadingAttributes.cs

```diff
@@ -1388,69 +1417,35 @@ .maxstack  8
 } // end of class Class1
 ";
 
-            var c1 = CreateStandardCompilation(
-@"
-public class Class1
+            var csSource = @"
+class Class2 : Class1
 {
-    public const decimal d1 = -7;
+}";
 
-    public void M1(decimal d1 = -7)
-    {}
-}");
+            CompileAndVerify(CreateCompilationWithCustomILSource(csSource, ilSource), symbolValidator: module =>
+            {
+                var class1 = module.GlobalNamespace.GetTypeMember("Class2").BaseType;
+                Assert.Equal("Class1", class1.ToTestDisplayString());
+
+                var field1 = class1.GetField("d1");
+
+                Assert.Empty(field1.GetAttributes());
+                Assert.Equal(field1.ConstantValue, -7m);
+
+                var field2 = class1.GetField("d2");
 
+                Assert.Equal(2, field2.GetAttributes().Length);
+                Assert.Null(field2.ConstantValue);
 
-            var class1 = c1.GetTypeByMetadataName("Class1");
-            var d1 = class1.GetMember<FieldSymbol>("d1");
-            var m1 = class1.GetMember<MethodSymbol>("M1");
-
-            var builder = GetDefaultPEBuilder(c1);
-
-            Assert.Empty(d1.GetAttributes());
-            Assert.Equal("System.Runtime.CompilerServices.DecimalConstantAttribute(0, 128, 0, 0, 7)", d1.GetCustomAttributesToEmit(builder).Single().ToString());
-            Assert.Equal(d1.ConstantValue, -7m);
-            Assert.Empty(m1.Parameters[0].GetAttributes());
-            Assert.Equal("System.Runtime.CompilerServices.DecimalConstantAttribute(0, 128, 0, 0, 7)", m1.Parameters[0].GetCustomAttributesToEmit(builder).Single().ToString());
-            Assert.Equal(m1.Parameters[0].ExplicitDefaultValue, -7m);
-
-            var c2 = CreateCompilationWithCustomILSource("", ilSource);
```

Comment:
>var c2 = CreateCompilationWithCustomILSource("", ilSource); [](start = 12, length = 59)

I think we should keep existing validation for symbols from c2 and c3. They are already PE symbols and it looks like the test asserts that API behavior is not order dependent.

Reply:
@AlekseyTs I see in the IL above that these members has duplicate `DecimalConstantAttribute` attributes. (0, 128, 0, 0, 7) and (0, 128, 0, 0, 8). Is this intended? I don't see an explanation in the bug linked in the test.

---
In reply to: [150639861](https://github.com/dotnet/roslyn/pull/22896#discussion_r150639861) [](ancestors = 150639861)